src/Native/System.Private.CoreLib.Native/configure.cmake

```diff
@@ -6,6 +6,26 @@ check_library_exists(pthread pthread_condattr_setclock "" HAVE_PTHREAD_CONDATTR_
 
 check_include_files(uuid/uuid.h HAVE_LIBUUID_H)
 
+check_cxx_source_compiles("
+#include <uuid/uuid.h>
+int main()
+{
+#ifndef uuid_generate_random
```

Comment:
This hack does not look like improvement. I would rather take the `WASM` ifdef that you had originally than this.

Reply:
Were you happy with the `check_function_exists` usage as below? This fails on OS X but I should imagine it is case of linking the correct library for the check. I don't have a OS X machine to test with at the moment, but if you bear with me I can try and sort something out. 

```
set(CMAKE_REQUIRED_LIBRARIES uuid)
check_function_exists(uuid_generate_random HAVE_UUID_GENERATE_RANDOM)
check_function_exists(uuid_generate HAVE_UUID_GENERATE)
set(CMAKE_REQUIRED_LIBRARIES)
```

Reply:
Yes, the `check_function_exists` usage looked much better.

Reply:
@ph1ll , if you need to commit and watch the CI results a couple times to figure out OS X, that's totally fine. I'll also try to get a build on a Mac soon, but it's probably easier for you to just iterate with CI.